File: packages/types/src/label.ts
export interface Label {
  id: string;
  name: string;
  color: string;
  projectId: string;
  createdAt: string;
}

export interface CreateLabel {
  name: string;
  color: string;
}

export interface UpdateLabel {
  name: string;
  color: string;
}

File: apps/Api/src/TestCraft.Domain/Entities/Label.cs
namespace TestCraft.Domain.Entities;

public class Label
{
    public LabelId Id { get; set; }
    public required string Name { get; set; }
    public required string Color { get; set; }
    public ProjectId ProjectId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }

    public Project? Project { get; set; }
    public ICollection<TestCaseLabel> TestCaseLabels { get; set; } = [];
}
